src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx

```diff
@@ -0,0 +1,401 @@
+      subtext =
+        activityDuration !== null &&
+        tct(`[currentTrigger]: [duration]`, {
+          currentTrigger,
+          duration: <Duration abbreviation seconds={activityDuration / 1000} />,
+        });
+    } else if (isClosed && incident?.statusMethod === IncidentStatusMethod.RULE_UPDATED) {
+      title = t('Alert auto-resolved');
+      subtext = t('Alert rule has been modified or deleted');
+    } else if (isClosed && incident?.statusMethod !== IncidentStatusMethod.RULE_UPDATED) {
+      title = t('Alert resolved');
+      subtext = tct('by [authorName]', {authorName});
+    } else if (isDetected) {
+      const nextActivity = activities.find(({previousValue}) => previousValue === '1');
+      const activityDuration = nextActivity
+        ? moment(nextActivity.dateCreated).diff(
+            moment(activity.dateCreated),
+            'milliseconds'
+          )
+        : null;
+
+      title = incident?.alertRule
+        ? t('Alert was created')
+        : tct('[authorName] created an alert', {authorName});
+      subtext =
+        activityDuration !== null &&
+        tct(`Critical: [duration]`, {
+          duration: <Duration abbreviation seconds={activityDuration / 1000} />,
+        });
+    } else if (isStarted) {
+      const dateEnded = moment(activity.dateCreated)
+        .add(rule.timeWindow, 'minutes')
+        .utc()
+        .format();
+      const timeOnly = Boolean(
+        dateEnded && moment(activity.dateCreated).date() === moment(dateEnded).date()
+      );
+
+      title = t('Trigger conditions were met');
+      subtext = (
+        <React.Fragment>
+          <DateTime
+            timeOnly={timeOnly}
+            timeAndDate={!timeOnly}
+            date={activity.dateCreated}
+          />
+          {' — '}
+          <DateTime timeOnly={timeOnly} timeAndDate={!timeOnly} date={dateEnded} />
+        </React.Fragment>
+      );
+    } else {
+      return null;
+    }
+
+    return (
+      <Activity key={activity.id}>
+        <ActivityTrack>
+          <IconEllipse size="sm" color="gray300" />
+          {!last && <VerticalDivider />}
+        </ActivityTrack>
+
+        <ActivityBody>
+          <ActivityTime>
+            <StyledTimeSince date={activity.dateCreated} suffix={t('ago')} />
+            <HorizontalDivider />
+          </ActivityTime>
+          <ActivityText>
+            {title}
+            {subtext && <ActivitySubText>{subtext}</ActivitySubText>}
+          </ActivityText>
+        </ActivityBody>
+      </Activity>
+    );
+  }
+
+  render() {
+    const {incident} = this.props;
+    const {activities} = this.state;
+
+    let Icon = IconCheckmark;
+    let color: string = theme.green300;
+
+    if (
+      activities &&
+      activities.find(
+        activity =>
+          activity.type === IncidentActivityType.DETECTED ||
+          (activity.type === IncidentActivityType.STATUS_CHANGE &&
+            activity.value === `${IncidentStatus.CRITICAL}`)
```

Comment:
could do `activities?.find`

There's something too long about these to be inside of the if statement

Reply:
or maybe filter by ```activity.type === IncidentActivityType.STATUS_CHANGE &&
            activity.value === `${IncidentStatus.CRITICAL}```` and then find on that

Reply:
I shortened it up a bit